src/algorand_transactions/transaction.rs

```diff
@@ -480,7 +511,11 @@ impl AlgorandTransaction {
                 )),
             },
             asset_transfer_transaction: self.to_asset_transfer_transaction_json(),
-            application_transaction: self.to_application_transaction_json(),
+            payment_transaction: self.to_payment_transaction_json(),
+            application_transaction: match self.to_application_transaction_json()? {
+                application_transaction if application_transaction.is_empty() => None,
+                application_transaction => Some(application_transaction),
+            },
```

Comment:
You can make this much more readable by having the `to_application_transaction_json` method return a `Result<Option<ApplicationTransactionJson>>`, leaving this line to look like:

```suggestion
            application_transaction: self.to_application_transaction_json()?,
```

That would make more sense than this match + match guard set up.

Reply:
Done